### PR TITLE
fix for #61 and tests for withdraw

### LINKED ISSUE #61: in HomeBridge.withdraw: recipient gets loaded from message with wrong offset and contains wrong data
code in question: https://github.com/paritytech/parity-bridge/blob/740601530096c261695ba9216b5e9d5f871a229f/contracts/bridge.sol#L119

look here for how message is constructed:
https://github.com/paritytech/parity-bridge/blob/59403966fc774bc3c0dc943b27d62f2c3ff8939a/bridge/src/bridge/withdraw_confirm.rs#L29

message format:
- 20 bytes recipient
- 32 bytes value
- 32 bytes transactionHash

the code currently mistakenly thinks the format is:
- **32** bytes recipient
- 32 bytes value
- 32 bytes transactionHash

all three values read in `HomeBridge.withdraw` are corrupted because of this.

found while writing tests for HomeBridge.withdraw (https://github.com/paritytech/parity-bridge/issues/60).

i've got a fix on the `issue-54-close-attack-vector` branch:
https://github.com/paritytech/parity-bridge/pull/56/commits/fed0cc1aaa4e4067fa3bde26aaa4c081cdc8897b

### PR DESCRIPTION
**Pull request by snd**
fixes #61 

includes some work on #60

### CODE REVIEW

File: truffle/test/helpers/helpers.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  var v = parseInt(signature.substr(64 * 2), 16);
+  var r = "0x" + signature.substr(0, 32 * 2);
+  var s = "0x" + signature.substr(32 * 2, 32 * 2);
+  return {v: v, r: r, s: s};
+}
+module.exports.signatureToVRS = signatureToVRS;
+
+// returns BigNumber `num` converted to a little endian hex string.
+// `num` must represent an unsigned integer
+function bigNumberToHexString(num) {
```
[the remark this answers is not in the document]
Comment (snd) [reply truffle/test/helpers/helpers.js]: i had some issues with `num.toString(16)` while figuring out bignums in truffle. seems to work though. i've just removed `bigNumberToHexString` in favour of `num.toString(16)`

**Comment by snd** [comment]
added some more tests for withdraw. done here. can i get another review?

**Comment by snd** [comment]
would have extracted the tests into their separate PR. they fail without the fix to #61 though

File: contracts/bridge.sol [hunk 2]
```diff
@@ -105,20 +105,41 @@ contract HomeBridge {
...
+            // if we were to read at 32 the address would contain part of value and be corrupted.
+            // when reading from offset 20 mload will read 12 zero bytes followed
+            // by the 20 recipient address bytes and correctly convert it into an address.
+            // this saves some storage/gas over the alternative solution
+            // which is padding address to 32 bytes and reading recipient at offset 32.
+            // for more details see discussion in:
+            // https://github.com/paritytech/parity-bridge/issues/61
+            recipient := mload(add(message, 20))
+            value := mload(add(message, 52))
+            hash := mload(add(message, 84))
```
Comment (debris) [inline contracts/bridge.sol]: thanks, for explanation, cause it's super confusing!

**Pull request merged**